algorithms/kernel/dtrees/forest/regression/oneapi/cl_kernels/df_batch_regression_kernels.cl

```diff
@@ -0,0 +1,994 @@
+                nodeList[nodeId * nNodeProp + 3]         = curFeatureValue == valNotFound ? -1 : curFeatureValue;
+                nodeList[nodeId * nNodeProp + 4]         = (int)bestLN;
+
+                splitNodeInfo[0] = ((algorithmFPType)0 != mrgN) ? mrgSum2Cent / mrgN : (algorithmFPType)0;
+                splitNodeInfo[1] = mrgMean;
+
+                if (updateImpDecreaseRequired) nodeImpDecreaseList[nodeId] = (algorithmFPType)0;
+                return;
+            }
+
+            if ((algorithmFPType)0 < impDec
+                && (curFeatureValue == -1 || impDec > curImpDec || (fpIsEqual(impDec, curImpDec) && featId < curFeatureId))
+                && mrgRN >= minObservationsInLeafNode && mrgLN >= minObservationsInLeafNode)
+            {
+                curFeatureId    = featId;
+                curFeatureValue = binId;
+                curImpDec       = impDec;
+
+                bestRN        = mrgRN;
+                bestRMean     = mrgRMean;
+                bestRSum2Cent = mrgRSum2Cent;
+
+                bestLN        = mrgLN;
+                bestLMean     = mrgLMean;
+                bestLSum2Cent = mrgLSum2Cent;
+            }
+        } // for i
+
+        algorithmFPType bestImpDec = sub_group_reduce_max(curImpDec);
+
+        int impDecIsBest     = fpIsEqual(bestImpDec, curImpDec);
+        int bestFeatureId    = sub_group_reduce_min(impDecIsBest ? curFeatureId : valNotFound);
+        int bestFeatureValue = sub_group_reduce_min((bestFeatureId == curFeatureId && impDecIsBest) ? curFeatureValue : valNotFound);
+
+        if (curFeatureId == bestFeatureId && curFeatureValue == bestFeatureValue)
+        {
+            if (1 == n_sub_groups)
+            {
+                __global algorithmFPType * splitNodeInfo = splitInfo + nodeId * nImpProp;
+                nodeList[nodeId * nNodeProp + 2]         = curFeatureId == valNotFound ? -1 : curFeatureId;
+                nodeList[nodeId * nNodeProp + 3]         = curFeatureValue == valNotFound ? -1 : curFeatureValue;
+                nodeList[nodeId * nNodeProp + 4]         = (int)bestLN;
+
+                splitNodeInfo[0] = ((algorithmFPType)0 != mrgN) ? mrgSum2Cent / mrgN : (algorithmFPType)0;
+                splitNodeInfo[1] = mrgMean;
+
+                if (updateImpDecreaseRequired) nodeImpDecreaseList[nodeId] = ((algorithmFPType)0 != mrgN) ? curImpDec / mrgN : (algorithmFPType)0;
+
+                return;
+            }
+            else
+            {
+                bufS[(bufIdx + sub_group_id) * nNodeProp + 0] = curFeatureId;
+                bufS[(bufIdx + sub_group_id) * nNodeProp + 1] = curFeatureValue;
+                bufS[(bufIdx + sub_group_id) * nNodeProp + 2] = (int)bestLN;
+
+                bufI[bufIdx + sub_group_id] = curImpDec;
+            }
+        }
+
+        barrier(CLK_LOCAL_MEM_FENCE);
+        if (1 < n_sub_groups && 0 == sub_group_id)
```

Comment:
Will never be called in case of a single subgroup?

Reply:
yes